cpp/include/cudf/io/datasource.hpp

```diff
@@ -335,26 +335,35 @@ class datasource {
   template <typename Container>
   class owning_buffer : public buffer {
    public:
+    // Disallow the container argument to be an lvalue (in which case Container = T& is a
+    // reference).
+    static_assert(!std::is_reference_v<Container>,
```

Comment:
should this be `is_lvalue_reference_v`?

Reply:
If the caller passes an lvalue argument, `Container` is deduced to be a reference (and also a lvalue reference); if the caller passes an rvalue, `Container` is a non-reference. So both `std::is_reference_v<Container>` and `std::is_lvalue_reference_v` would work in this case.

Equivalently, we can have:
```
static_assert(!std::is_lvalue_reference_v<Container&&>, "...");
```

Reply:
I think something that involves lvalue would be clearer. The original was not obvious to me.

Reply:
Completed, using `is_rvalue_reference_v<T&&>` to further improve clarity.